qiskit/pulse/utils.py

```diff
@@ -114,3 +115,28 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
 
     return wrapper
+
+
+def lambdify_symbolic_pulse(
+    expr: sympy.Expr,
+    parameters: List[str],
+) -> Callable:
+    """Lambdify symbolic pulse equation.
+
+    Args:
+        expr: Sympy or symengine symbolic equation of the pulse shape.
+        parameters: Name list of pulse parameters.
+
+    Returns:
+        Lambdified symbolic equation.
+    """
+    symbols = [sympy.Symbol(p) for p in parameters]
+
+    # if optionals.HAS_SYMENGINE:
+    #     # Fall back to sympy for lambda function creation until symengine's
+    #     # lambdify fully supports the features required by parametric
+    #     # pulses.
+    #     import symengine
+    #     expr = symengine.sympify(expr)
```

Comment:
We might want to leave a symengine version here in a comment for reference for once piece-wise support is available. In https://github.com/nkanazawa1989/qiskit-terra/pull/48, I had mentioned that symengine did not support complex numbers. It turns out it does, but you need to pass `real=False` to `Lambdify`.

So `sympy.lambdify(symbols, expr)` becomes `symengine.lambify(symbols, [expr], real=False)`.